File: homework/single_precision.cc
#include <iostream>
#include <math.h>

using namespace std;

int main() {
    float x = 1;
    float y = 1;
    float z = x +1;
    int time = 0;
    
    while (z != 1){
        x = x/2.0;
        z =  y + x;
        time = time +1.0;
        if ((z - y) == 0.0){
            break;
        }
    }
    cout << "ival: " << x << endl << "ipow: " << time << endl;
    return 0;
}

File: homework/Single_Precision.m
x = single(1);
y = single(2);
z = single(1);
time = 0;

while (x+z) ~= single(1)
    z = single(z/y);
    x = single(1) +z;
    time = time +1
    if time == 100000
        break
    end
end